pkgs/applications/science/biology/cmtk/default.nix

```diff
@@ -1,31 +1,50 @@
-{ lib, stdenv, fetchurl, cmake }:
+{ lib
+, stdenv
+, bzip2
+, cmake
+, fetchurl
+, fftw
+, llvmPackages
+, zlib
+}:
 
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
   pname = "cmtk";
-  version = "3.3.1";
+  version = "3.3.2";
 
   src = fetchurl {
     name = "cmtk-source.tar.gz";
-    url = "https://www.nitrc.org/frs/download.php/8198/CMTK-${version}-Source.tar.gz//?i_agree=1&download_now=1";
-    sha256 = "1nmsga9m7vcc4y4a6zl53ra3mwlgjwdgsq1j291awkn7zr1az6qs";
+    url = "https://www.nitrc.org/frs/download.php/13188/CMTK-${finalAttrs.version}-Source.tar.gz//?i_agree=1&download_now=1";
+    hash = "sha256-iE164NCOSOypZLLZfZy9RTyrS+YnY9ECqfb4QhlsMS4=";
   };
 
+  patches = [
+    ./replace-finite-with-isfinite.patch
```

Comment:
Can the patch be replaced by `substituteInPlace` or `-Dfinite=isfinite` in `env.NIX_CFLAGS_COMPILE`?

Reply:
Thanks for your suggestion.
I had given up on using `substituteInPlace` because it was difficult here, but `-Dinfinite=isinfinite` worked fine.